baremetal/metal3machine_manager.go

```diff
@@ -1729,9 +1748,14 @@ func (m *MachineManager) getNodesWithLabel(ctx context.Context, nodeLabel string
 		return nil, 0, errors.Wrap(err, "Error creating a remote client")
 	}
 	nodesCount := 0
-	nodes, err := corev1Remote.Nodes().List(ctx, metav1.ListOptions{
+	filter := metav1.ListOptions{
 		LabelSelector: nodeLabel,
-	})
+	}
+	// When no labels are in use, retrieve all nodes
+	if !m.Metal3Cluster.Spec.NoCloudProvider {
+		filter = metav1.ListOptions{}
+	}
```

Comment:
Lines 1751-1757 isn't clear to me, is it so that if NoCloudProvier is false, we return all the nodes ? and Why is that ?

Reply:
Yes. 
If NoCloudProvier=false, we get all nodes. The nodes could have labels or not, it should not matter. 
However, 
If NoCloudProvier=true, then we just get the nodes with labels.